### ISSUE TITLE
Display the current filename somewhere in the UI

### ISSUE THREAD

**Issue by sompylasar** [excalidraw/excalidraw#1674]
RE: #1668 I just realized that to help the user decide whether they need to hit "Save" or "Save as" button, we might want to display the current filename somewhere in the UI if it's not there already -- I don't remember seeing it -- maybe browser tab title on desktop, not sure about mobile.

CC @tomayac

**Comment by tomayac** [comment]
Please see [`/src/actions/actionExport.tsx`](https://github.com/excalidraw/excalidraw/blob/f415a52ee7631c04a7522aaa8fc0fa423e1c7061/src/actions/actionExport.tsx) from https://github.com/excalidraw/excalidraw/pull/1709. My idea was something along the lines of…

```jsx
<ToolButton
    type="button"
    icon={save}
    title={t("buttons.save")}
    aria-label={t("buttons.save")}
    showAriaLabel={useIsMobile()}
    onClick={() => updateData(null)}
    onPointerOver={() => {
      this.props.title = `${
        (window as any).handle
        ? ' (' + (window as any).handle.name + ')'
        : ''}`
    }
/>
```

…but I'm too much of a React n00b 🙈 and this wouldn't compile. I made `ToolButton` aware of `onPointerOver`, but didn't know how to access `this` from within the function.

Anyway, the idea essentially would be to dynamically change the title attribute `onpointerover` and make it show the current file name `(window as any).handle.name`. Help wanted.

**Comment by chenjefferson** [comment]
Hello! Working on this task (unless someone tells me it's not ideal).

**Comment by yogeshwara-krishna** [comment]
@sompylasar @tomayac Hello, is this issue still open? can I be assigned to it? would like to try as a good first issue.

**Comment by dwelle** [comment]
Hey @yogeshwara-krishna, thanks! I've assigned you.

To spec this out a bit more, let's do this:

1. show the filename next to the main menu button like this:

    ![image](https://github.com/excalidraw/excalidraw/assets/5153846/e064edb7-dc8c-4cdc-abc5-01b90e4ecc4b)

2. only show this when the `appState.fileHandle` is defined and use its file name
3. on click, save back to the file

**Comment by yogeshwara-krishna** [comment]
@dwelle Hello, I have a doubt, how should it be displayed in the mobile view screen?

**Comment by dwelle** [comment]
@yogeshwara-krishna good question. I'd either not display it anywhere, or suffix it to the `save to current file` menu item:

![image](https://github.com/excalidraw/excalidraw/assets/5153846/f8fb6f22-7ce4-4b47-9018-c2ccc6e53fdf)

But it would almost always not fit, so perhaps it's fine not to put it anywhere on mobile.

**Comment by tomayac** [comment]
On mobile, the problem doesn’t arise, since you can’t save to the same file, but need to download to a new file. This is because the File System Access API doesn’t exist on mobile. So on mobile it doesn’t make sense to show a file name.

**Comment by yogeshwara-krishna** [comment]
@dwelle @tomayac Hello, I have made a PR, please review it. Feedback and suggestions are welcome 😄

**Comment by Ayush-Porwal** [comment]
@tomayac you can also review #6985 , I haven't touched the mobile screen

**Comment by KennyClayton** [comment]
So this feature was never implemented?

I can inspect and see it here:

![Image](https://github.com/user-attachments/assets/24c187e7-2ab2-420a-af90-68813783d96c)

**Comment by ayush-porwal** [comment]
> So this feature was never implemented?
> 
> I can inspect and see it here, which is my workaround right now:
> 
> ![Image](https://github.com/user-attachments/assets/24c187e7-2ab2-420a-af90-68813783d96c)

Nope, I also did a pr #6985 for the same issue. That was also not merged.

**Comment by MuhammadSahimBhaur** [comment]
@sompylasar shouldn't this be closed? This whole board seems outdated. Is this still active?

**Comment by Nileshkethavath** [comment]
@dwelle is this still open ?

**Comment by Dunkrick** [comment]
Is this issue still open? I would like to work on it

**Comment by MathiasLui** [comment]
This has bothered me for a couple years, it doesn't seem like it's been merged, which is surprising as it's such a basic feature in every program. 

If I hit "Save" i wanna be sure that I don't override a different file (which has happened before). Because of this issue, I only use "Save to..."

**Comment by Devanshi-Vadiya** [comment]
hey there is this issue still open??

**Comment by sathish-chanda** [comment]
Hello @dwelle, 

I am not sure if this implemented. so, I have implemented it. Please could you review my  https://github.com/excalidraw/excalidraw/pull/12139 MR.

<img width="1496" height="838" alt="Image" src="https://github.com/user-attachments/assets/5f41c68e-a157-4be6-a11f-f3e8f21155b2" />

<img width="1499" height="837" alt="Image" src="https://github.com/user-attachments/assets/d483db87-6667-43d3-b0cc-3dfff14b2407" />
